main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -1075,6 +1075,65 @@ object Typer extends Phase[ResolvedAst.Program, TypedAst.Root] {
             inferredArgumentTypes <- seqM(actuals.map(visitExp))
           ) yield tvar
 
+        /*
+         * New Channel expression.
+         */
+        case ResolvedAst.Expression.NewChannel(tvar, loc) =>
+          unifyM(tvar, Type.mkChannel(Type.freshTypeVar()), loc)
+
+        /*
+         * Get Channel expression.
+         */
+        case ResolvedAst.Expression.GetChannel(exp, tvar, loc) =>
+          for {
+            actualExpType <- visitExp(exp)
+             _ <- unifyM(actualExpType, Type.mkChannel(tvar), loc)
+          } yield tvar
+
+        /*
+         * Put Channel expression.
+         */
+        case ResolvedAst.Expression.PutChannel(exp1, exp2, tvar, loc) =>
+          val elementType = Type.freshTypeVar()
+          for {
+            e1 <- visitExp(exp1)
+            e2 <- visitExp(exp2)
+            _ <- unifyM(e1, Type.mkChannel(elementType), loc)
+            _ <- unifyM(e2, elementType, loc)
+            resultType <- unifyM(tvar, Type.mkChannel(elementType), loc)
+          } yield resultType
```

Comment:
May I propose the following variable name changes:
elementType -> freshElementTypeVar
e1 -> channelType
e2 -> elementType

Reply:
Fixed